components/layout_thread/lib.rs

```diff
@@ -1111,8 +1111,8 @@ impl LayoutThread {
 
                         let mut current = node.parent_node();
                         while let Some(node) = current {
-                            if node.has_dirty_descendants() { break; }
-                            unsafe { node.set_dirty_descendants(); }
+                            if node.is_element() && node.as_element().unwrap().has_dirty_descendants() { break; }
```

Comment:
Given we're checking on `parent_node`, this is guaranteed to be an element, right?